connection_go18.go

```diff
@@ -0,0 +1,28 @@
+// +build go1.8
+
+package mysql
+
+import (
+	"context"
+	"time"
+)
+
+func (mc *mysqlConn) Ping(ctx context.Context) error {
+	err := mc.writeCommandPacket(comPing)
+	if err != nil {
+		return err
+	}
+
+	ch := make(chan error)
+	go func() {
+		_, err := mc.readResultOK()
+		ch <- err
+	}()
+	select {
+	case <-ctx.Done():
+		mc.netConn.SetReadDeadline(time.Now())
+		return ctx.Err()
```

Comment:
This is dangerous.

* mc is not thread safe.
* `mc.readResultOK()` is still working. 
* After `Ping` return, `database/sql.DB` will use this connection again. race may be happen.